### DISCUSSION TITLE
Questions about istio - consul integration.

### DISCUSSION THREAD

**Discussion by kimmikimmi** [istio/istio#51490] [Q&A] [answered]
Hi team! I have several questions about Consul integration using MCP over XDS.

I am considering to integrate consul only for an endpoint discovery part. and checked both Consul as XDS server and configcontroller of Istio. But I encountered issues below.
1.  that MCP does not support DELTA interface, only `StreamAggregatedResources` is supported (https://github.com/istio/istio/blob/c804513f1844ad289a557a48704c71070a2c4d80/pilot/pkg/bootstrap/configcontroller.go#L230)
2.  Consul deprecated SotW ads (only delta), Consul is able to sync kube-api servers for all the endpoints within cluster. (https://github.com/hashicorp/consul/blob/ffa7aff2073d91cd68559519be6a3ab64272dd1f/agent/xds/server.go#L169)

Q1. Do you plan to support Delta interface on MCS over XDS? I saw [Remove obsolete MCP APIs](https://github.com/istio/api/pull/3211) PR here. is it going to be retired in the future?
Q2. if not, Is there recommended ways to integrate Istio with Consul? 
Do I need to build my own Istio-Consul adapter(Custom MCP server) to achieve this? (e.g. https://github.com/costinm/istio-consul). 

it will be very helpful if you guys can share thoughts and direction of istio.

Many Thxx.

**Comment by howardjohn** [comment, marked as answer]
There is no MCP API at all, in any form, in Istio currently for many years. There is only XDS, programmed using the pkg/adsc library. Any references to MCP are effectively deprecated.

ADSC has a delta mode (https://github.com/istio/istio/blob/7227aeec01d3fc54c15872a8e27c4b250647ee2b/pkg/adsc/delta.go#L330) but its not yet integrated into the main Istiod. If someone wanted to do the work and maintain this it would be great.

AFAIK most users integrating with consul are using https://github.com/hashicorp/consul-k8s, though I am not an expert

**Comment by kimmikimmi** [comment]
Thanks for reply! very helpful. will consider to contribute this part.
